### ISSUE TITLE
problem with unsubscribe ratio as filter in reports

### ISSUE THREAD

**Issue by johbuch** [mautic/mautic#7054]
[//]: # ( Invisible comment: 
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Before you create the issue:
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Search for similar report among other reported issues.
Learn how to troubleshoot at https://www.mautic.org/docs/en/tips/troubleshooting.html
Use drag&drop to attach images or other files )

## Bug Description
If you add the unsubscribe ratio (or bounce ratio) as a filter in a email report, it generates a 500 error.

| Q   | A
| --- | ---
| Mautic version | 2.15
| PHP version | 
| Browser | 

### Steps to reproduce
1. Create a report and choose email as data source
![image](https://user-images.githubusercontent.com/31535432/50449423-9df58a80-0927-11e9-94e3-ec8ba3978c69.png)
2. add some columns.
3. Save. See that there is no problem with the report
4. Now go back editing the report and add unsubscribe ratio (or bounce ratio) as a filter. Choose greater than and choose a number
![image](https://user-images.githubusercontent.com/31535432/50449466-deed9f00-0927-11e9-851a-a7a06f46c949.png)
5. Save and close. See error 500
![image](https://user-images.githubusercontent.com/31535432/50449474-ee6ce800-0927-11e9-8890-4f026d1fcf20.png)
 
### Log errors
Here are the logs related to the 500 error
```
[2018-12-26 15:41:22] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\DriverException: "An exception occurred while executing 'SELECT count(*) FROM (SELECT e.id AS e_id1, e.name AS e_name1, e.subject AS subject2, CONCAT(ROUND((SUM(IF(dnc.id IS NOT NULL AND dnc.reason=2 , 1, 0))/e.sent_count)*100, 1),'%') AS bounced_ratio1, CONCAT(ROUND(cut.hits/(e.sent_count)*100, 1),'%') AS hits_ratio1, CONCAT(ROUND((e.read_count/e.sent_count)*100, 1),'%') AS read_ratio1, CONCAT(ROUND(cut.unique_hits/(e.sent_count)*100, 1),'%') AS unique_ratio1, CONCAT(ROUND((SUM(IF(dnc.id IS NOT NULL AND dnc.reason=1 , 1, 0))/e.sent_count)*100, 1),'%') AS unsubscribed_ratio1 FROM emails e LEFT JOIN emails vp ON vp.id = e.variant_parent_id LEFT JOIN categories c ON c.id = e.category_id LEFT JOIN (SELECT COUNT(cut2.channel_id) AS trackable_count, SUM(cut2.hits) AS hits, SUM(cut2.unique_hits) AS unique_hits, cut2.channel_id FROM channel_url_trackables cut2 WHERE cut2.channel = 'email' GROUP BY cut2.channel_id) cut ON e.id = cut.channel_id LEFT JOIN lead_donotcontact dnc ON e.id = dnc.channel_id AND dnc.channel='email' WHERE (e.date_added IS NULL OR (e.date_added BETWEEN ? AND ?)) AND (CONCAT(ROUND((SUM(IF(dnc.id IS NOT NULL AND dnc.reason=1 , 1, 0))/e.sent_count)*100, 1),'%') >= ?) GROUP BY e.id) c' with params ["2018-09-25 22:00:00", "2018-12-26 14:41:22", "20"]:  SQLSTATE[HY000]: General error: 1111 Invalid use of group function" at /home/atmt_yyymaut-8/yyymaut-8.automation.webmecanik.com/mautic/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 115 {"exception":"[object] (Doctrine\\DBAL\\Exception\\DriverException(code: 0): An exception occurred while executing 'SELECT count(*) FROM (SELECT e.id AS e_id1, e.name AS e_name1, e.subject AS subject2, CONCAT(ROUND((SUM(IF(dnc.id IS NOT NULL AND dnc.reason=2 , 1, 0))/e.sent_count)*100, 1),'%') AS bounced_ratio1, CONCAT(ROUND(cut.hits/(e.sent_count)*100, 1),'%') AS hits_ratio1, CONCAT(ROUND((e.read_count/e.sent_count)*100, 1),'%') AS read_ratio1, CONCAT(ROUND(cut.unique_hits/(e.sent_count)*100, 1),'%') AS unique_ratio1, CONCAT(ROUND((SUM(IF(dnc.id IS NOT NULL AND dnc.reason=1 , 1, 0))/e.sent_count)*100, 1),'%') AS unsubscribed_ratio1 FROM emails e LEFT JOIN emails vp ON vp.id = e.variant_parent_id LEFT JOIN categories c ON c.id = e.category_id LEFT JOIN (SELECT COUNT(cut2.channel_id) AS trackable_count, SUM(cut2.hits) AS hits, SUM(cut2.unique_hits) AS unique_hits, cut2.channel_id FROM channel_url_trackables cut2 WHERE cut2.channel = 'email' GROUP BY cut2.channel_id) cut ON e.id = cut.channel_id LEFT JOIN lead_donotcontact dnc ON e.id = dnc.channel_id AND dnc.channel='email' WHERE (e.date_added IS NULL OR (e.date_added BETWEEN ? AND ?)) AND (CONCAT(ROUND((SUM(IF(dnc.id IS NOT NULL AND dnc.reason=1 , 1, 0))/e.sent_count)*100, 1),'%') >= ?) GROUP BY e.id) c' with params [\"2018-09-25 22:00:00\", \"2018-12-26 14:41:22\", \"20\"]:\n\nSQLSTATE[HY000]: General error: 1111 Invalid use of group function at /home/atmt_yyymaut-8/yyymaut-8.automation.webmecanik.com/mautic/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:115, Doctrine\\DBAL\\Driver\\PDOException(code: HY000): SQLSTATE[HY000]: General error: 1111 Invalid use of group function at /home/atmt_yyymaut-8/yyymaut-8.automation.webmecanik.com/mautic/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:107, PDOException(code: HY000): SQLSTATE[HY000]: General error: 1111 Invalid use of group function at /home/atmt_yyymaut-8/yyymaut-8.automation.webmecanik.com/mautic/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:105)"} []
``` 
[//]: # ( Invisible comment:
Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable. )

**Comment by kuzmany** [comment]
Try test https://github.com/mautic/mautic/pull/7070

**Comment by npracht** [comment]
Merged in 2.15.1
